### PR TITLE
Logic of publisher_publish command moved to cms.api

### PR DESCRIPTION
**Pull request by donce**
Solves #3279.

### CODE REVIEW

**Comment by donce** [comment]
Hmm, do you know what could have caused this error?

File: cms/api.py [hunk 1]
```diff
@@ -503,3 +504,36 @@ def copy_plugins_to_language(page, source_language, target_language,
...
     return copied
+
+
+def publish_pages(include_unpublished, language, site):
+    """
+    Create published public version of selected drafts.
+    """
+    from cms.models import Page
+
+    # thread locals middleware needs to know, who are we - login as a first
```
Comment (yakky) [inline cms/api.py]: Comment does not belong here, it seems

File: cms/api.py [hunk 2]
```diff
@@ -503,3 +504,36 @@ def copy_plugins_to_language(page, source_language, target_language,
             copied_plugins = copy_plugins.copy_plugins_to(plugins, placeholder, target_language)
             copied += len(copied_plugins)
     return copied
+
+
+def publish_pages(include_unpublished, language, site):
+    """
+    Create published public version of selected drafts.
+    """
+    from cms.models import Page
```
Comment (yakky) [inline cms/api.py]: Page models should be already available in api

**Comment by yakky** [comment]
Probably just travis hiccup, test restarted.
Other than those two minor comments LGTM
This is probably a good place to move the command itself as subcommand of cms command, to streamline the API. Would you do that?

**Comment by donce** [comment]
Sure, I'll make it as a cms subcommand :)

**Comment by donce** [comment]
Also, thanks for comments - I'll fix them later today/tomorrow :)

**Comment by yakky** [comment]
@donce awesome!

**Comment by yakky** [comment]
Would you add documentation for this function to https://raw.githubusercontent.com/divio/django-cms/develop/docs/reference/api_references.rst just like the sibling functions?

**Comment by donce** [comment]
Added docs, made some small code changes and rebased :)

**Comment by yakky** [comment]
@donce thanks, one last bit of effort, sorry: could you move publisher_publish to subcommands? That would fix a long standing issue and would be a really nice addition to this

**Comment by donce** [comment]
Of course, I've already started doing that :P But I'll finish it another day :)

**Comment by yakky** [comment]
Great, thanks! I changed the title to remember not to merge before you're done

**Comment by yakky** [comment]
Sorry, I've not been notified you completed this.
Really nice job!

**Pull request merged**